### ISSUE TITLE
Implement Groups API

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#671]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

In addition to the organizations API for #627, we also need a similar API for groups that are sub-organizations. Many of the model elements will end up being the same, but there are some minor differences. The schemas for groups can be seen [on Figma here](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=5377%3A96660&mode=design&t=GLVfWMR4HNeoHP0L-1).

### Contribution

Happy to support on this!

**Comment by andrewtavis** [comment]
CC @to-sta :)

**Comment by kurtisolo** [comment]
Unfortunately, as things stand, I don't have enough free time to delve into the intricate world of the project's APIs. However, I can suggest an approach that could simplify the work as the various `ViewSets` are developed. Certainly, as development continues, the files will start to become huge again and at some point difficult to navigate. An approach that I use in my platform, which makes my work much easier, is to divide the apps into sub-folders.
Practical example:
```
entities/
  |
  +--groups/
    |
    +--__init__.py
    +--views.py
    +--serializers.py
    +--whatever_else.py
  |
  +--organizations/
    |
    +--__init__.py
    +--views.py
    +--serializers.py
    +--whatever_else.py
  |
  +--__init__.py
  +--admin.py
  +--models.py
  +--urls.py
  +--etc.py
```
This way, navigating the backend would be much simpler, immediate, and intuitive. This would not entail migrations or other structural problems; it's simply a way to better organize the code. This is a long-term approach: surely at some point the various `views.py` files will become huge, so it might be wise to nip the problem in the bud.

My two cents =)

**Comment by andrewtavis** [comment]
Yes splitting it up where we can makes sense, @kurtisolo! Thanks for the suggestion :)

**Comment by andrewtavis** [comment]
Good chance that myself or @to-sta will work on this, so we'll keep this in mind 😊

**Comment by thiagogquinto** [comment]
Hey @andrewtavis, could I try to work on this one?

**Comment by andrewtavis** [comment]
Hey @thiagogquinto 👋 @to-sta and I checked in and it'd be great to have you work on this one. Let us know if there's anything you need!

**Comment by thiagogquinto** [comment]
Hey @andrewtavis, I checked the schema for the Group table and the models.py files is out of date from figma, and while I was checking I noticed that there are other models that need an update. In that way, I thought in first make an issue for a full update of the models as well the serializers and then proceeding to this issue. The update would change only the already existing tables, so the fields that need new tables would not be added. Please, let me know what you think!

**Comment by andrewtavis** [comment]
Sounds good to me, @thiagogquinto! 😊

**Comment by andrewtavis** [comment]
I'll assign once the new issue is made :)

**Comment by thiagogquinto** [comment]
Hey @andrewtavis and @to-sta, I brought to the issue branch the latest changes. Although that when I run the envinroment show that the the models ``Status`` and ``StatusType`` were created, when I try to test the API, occurs an error about the field ``entities_organization.status_id`` does not exist:

```sql
postgres_db     | 2024-03-19 13:58:17.024 UTC [57] ERROR:  column entities_organization.status_id does not exist at character 304
postgres_db     | 2024-03-19 13:58:17.024 UTC [57] STATEMENT:  SELECT "entities_organization"."id", "entities_organization"."name", "entities_organization"."tagline", "entities_organization"."org_icon_id", "entities_organization"."created_by_id", "entities_organization"."description", "entities_organization"."social_accounts", "entities_organization"."high_risk", "entities_organization"."status_id", "entities_organization"."status_updated", "entities_organization"."acceptance_date", "entities_organization"."deletion_date" FROM "entities_organization" WHERE "entities_organization"."id" = '3fa85f64-5717-4562-b3fc-2c963f66afa6'::uuid LIMIT 21
```

 Then when I try to check the table through _/v1/entities/status/_ there is the following error message: ``relation "entities_status" does not exist
LINE 1: SELECT COUNT(*) AS "__count" FROM "entities_status"``

**Comment by andrewtavis** [comment]
Closed via #783. Thanks for all the work to get the backend up and running, @thiagogquinto! We really do appreciate the dedication to the project 😊
